src/Dependencies/PooledObjects/PooledStringBuilder.cs

```diff
@@ -72,10 +72,18 @@ public string ToStringAndFree(int startIndex, int length)
         private static readonly ObjectPool<PooledStringBuilder> s_poolInstance = CreatePool();
 
         // if someone needs to create a private pool;
-        public static ObjectPool<PooledStringBuilder> CreatePool()
+        /// <summary>
+        /// If someone need to create a private pool
+        /// </summary>
+        /// <param name="size">Initial size of the pool. Bounded to 32 &lt;= Size &lt;= 256)</param>
+        /// <returns></returns>
+        public static ObjectPool<PooledStringBuilder> CreatePool(int size = 32)
         {
+            const int lower_bound = 32;
+            const int upper_bound = 256;
```

Comment:
>const int upper_bound = 256; [](start = 12, length = 28)

Why do we need to check bounds in this function at all? We should simply pass the size to the constructor of the ObjectPool and let it do any validation that that class care about.